tardis/plasma/tests/test_nlte_solver.py

```diff
@@ -218,3 +224,64 @@ def test_jacobian_matrix(
         [0.0, 1.0, 0.0, 1.0, 2.0, -1.0],
     ]
     assert_almost_equal(actual_jacobian_matrix, desired_jacobian_matrix)
+
+
+@pytest.fixture(scope="session")
+def nlte_atomic_data_fname(tardis_ref_path):
+    atomic_data_fname = os.path.join(
+        tardis_ref_path, "nlte_atom_data", "TestNLTE_He_Ti.h5"
+    )
+
+    atom_data_missing_str = (
+        f"{atomic_data_fname} atomic datafiles " f"does not seem to exist"
+    )
+
+    if not os.path.exists(atomic_data_fname):
+        pytest.exit(atom_data_missing_str)
+
+    return atomic_data_fname
+
+
+@pytest.fixture(scope="session")
+def nlte_atomic_dataset(nlte_atomic_data_fname):
+    nlte_atomic_data = AtomData.from_hdf(nlte_atomic_data_fname)
+
+    # if atomic_data.md5 != DEFAULT_ATOM_DATA_UUID:
+    #     pytest.skip(
+    #         f'Need default Kurucz atomic dataset (md5="{DEFAULT_ATOM_DATA_UUID}")'
+    #     )
+    # else:
+    return nlte_atomic_data
+
+
+@pytest.fixture
+def nlte_atom_data(nlte_atomic_dataset):
+    return nlte_atomic_dataset.atom_data
+
+
+data_path = os.path.join("tardis", "io", "tests", "data")
+
+
+@pytest.fixture
+def tardis_model_nlte():
+    filename = "tardis_configv1_nlte.yml"
+    return Configuration.from_yaml(os.path.join(data_path, filename))
+
+
+@pytest.fixture
+def nlte_raw_model(tardis_model_nlte):
+    return Radial1DModel.from_config(tardis_model_nlte)
+
+
+@pytest.fixture
+def nlte_raw_plasma(tardis_model_nlte, nlte_raw_model, nlte_atom_data):
+    plasma = assemble_plasma(tardis_model_nlte, nlte_raw_model, nlte_atom_data)
+    plasma.update(w=1.0)
+    return plasma
+
+
+def test_critical_case(nlte_raw_plasma):
+    assert_almost_equal(
+        nlte_raw_plasma.ion_number_density,
+        nlte_raw_plasma.ion_number_density_nlte,
```

Comment:
At some point soon, we want the Solver to return just `ion_number_density` (not `ion_number_density_nlte`). We also don't want to calculate the LTE solution by default. We should think about how to restructure the test to take these points into account. For calculating the LTE solution in the test, we could make use of the fact that `PhiSahaLTE` is computed by default (since it is needed for the continuum processes in the macro atom).

Reply:
I can save the values ran with nlte and update the plasma to not have any nlte species and then compare them?